src/seedu/addressbook/commands/SortCommand.java

```diff
@@ -0,0 +1,32 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+
+import java.util.Collections;
+import java.util.List;
+
+
+/**
+ * Displays sorted list of persons to the user.
+ */
+
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE  = COMMAND_WORD + ": Displays all persons in the address book sorted in "
+            + "alphabetical order.\n"
+            + "Example: " + COMMAND_WORD;
+
+    public SortCommand() {}
+
+    @Override
+    public CommandResult execute() {
+        UniquePersonList allPersons = addressBook.getAllPersons();
+        allPersons.sort();
```

Comment:
From the `AddressBook` class's getAllPersons method:

> Returns a new UniquePersonList of all persons in the address book at the time of the call.

This allPersons is thus not the same as the allPersons in the main AddressBook, thus while your sorting is correct, it does not persist across calls. Try calling `sort` then `list`.